src/report/errors.rs

```diff
@@ -106,13 +109,28 @@ impl Errors {
         false
     }
 
+    fn should_log_once(&mut self, report: &LogReport) -> bool {
+        if self.log_once.contains(&report.key) {
+            for (stored_report, count) in &mut self.storage {
+                if stored_report.eq_locless(report) {
+                    *count += 1;
+                    return true;
+                }
```

Comment:
I'm worried about the cost of this. Scanning the whole log again for every new entry makes building up the error log quadratic in time. Some mods do have tens of thousands of reports.

Reply:
> Would it? As far as I can tell all `LogReport`'s are instantiated through the builder. Should just be able to update `ReportBuilderStage3.build()` and `ReportBuilderStage3.push()` without touching any other code

I'm not sure what you have in mind, but if you say so :)

> 🤔 Would this require maintaining a second hashset? Or is there a way to cast `TigerHashSet<LogReport>` to `TigerHashSet<LogReportWrapper>` without copying?

It would mean having a `TigerHashMap<LogReportWrapper,TigerHashSet<LogReport>>`, similar to one of your proposals.